internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	// Validate allowed_origins value(s)
+	if internaltypes.IsDefined(model.AllowedOrigins) {
+		aoElems := model.AllowedOrigins.Elements()
+		for _, aoElem := range aoElems {
+			aoElemString := aoElem.(basetypes.StringValue).ValueString()
+			isElemUrl := internaltypes.IsUrlFormat(aoElemString)
+			if !isElemUrl {
+				resp.Diagnostics.AddError("Invalid URL Format!", fmt.Sprintf("Please provide a valid origin. Origin \"%s\" needs to be in a valid URL-like format - \"http(s)//:<value>.<domain>\"", aoElemString))
+			}
+		}
+	}
+
+	// Scope list for comparing values in matchNameBtwnScopes variable
+	scopeNames := []string{}
+
+	// Test scope names for dynamic true, string must be prepended with *
+	if internaltypes.IsDefined(model.Scopes) {
+		scopeElems := model.Scopes.Elements()
+		for _, scopeElem := range scopeElems {
+			scopeElemObjectAttrs := scopeElem.(types.Object)
+			scopeEntryName := scopeElemObjectAttrs.Attributes()["name"].(basetypes.StringValue).ValueString()
+			scopeNames = append(scopeNames, scopeEntryName)
+			scopeEntryIsDynamic := scopeElemObjectAttrs.Attributes()["dynamic"].(basetypes.BoolValue).ValueBool()
+			if scopeEntryIsDynamic {
+				if string(scopeEntryName[0]) != "*" {
+					resp.Diagnostics.AddError("Scope name conflict!", fmt.Sprintf("Scope name \"%s\" must be prefixed, suffixed or both by static part(s) with one and only one dynamic part represented by \"*\"", scopeEntryName))
+				}
+			}
+		}
+	}
+
+	// Test exclusive scope names for dynamic true, string must be prepended with *
+	eScopeNames := []string{}
+	if internaltypes.IsDefined(model.ExclusiveScopes) {
+		exclusiveScopeElems := model.ExclusiveScopes.Elements()
+		for _, esElem := range exclusiveScopeElems {
+			esElemObjectAttrs := esElem.(types.Object)
+			eScopeEntryName := strings.ToLower(esElemObjectAttrs.Attributes()["name"].(basetypes.StringValue).ValueString())
+			eScopeNames = append(eScopeNames, eScopeEntryName)
+			eScopeEntryIsDynamic := esElemObjectAttrs.Attributes()["dynamic"].(basetypes.BoolValue).ValueBool()
+			if eScopeEntryIsDynamic {
+				if string(eScopeEntryName[0]) != "*" {
+					resp.Diagnostics.AddError("Dynamic scope name conflict!", fmt.Sprintf("Dynamic scope name \"%s\" must be prefixed, suffixed or both by static part(s) with one and only one dynamic part represented by \"*\"", eScopeEntryName))
+				}
+			}
+		}
+	}
+
+	// Test if values in sets match
+	matchNameBtwnScopes, matchVal := internaltypes.MatchStringInSets(scopeNames, eScopeNames)
+	if matchNameBtwnScopes {
+		resp.Diagnostics.AddError("Scope name conflict!", fmt.Sprintf("The scope name \"%s\" is already defined in another scope list", matchVal))
+	}
+}
+
+func addOptionalOauthAuthServerSettingsFields(ctx context.Context, addRequest *client.AuthorizationServerSettings, plan oauthAuthServerSettingsResourceModel) error {
+
+	if internaltypes.IsDefined(plan.DefaultScopeDescription) {
```

Comment:
It looks like some of the attributes in this method such as this one aren't optional, so we don't need to re-add them to the addrequest here.

Reply:
Add unnecessary requests were removed in the latest commit.